code/game/objects/micro_structures.dm

```diff
@@ -0,0 +1,201 @@
+/obj/structure/micro_tunnel
+	name = "mouse hole"
+	desc = "A tiny little hole... where does it go?"
+	icon = 'icons/obj/structures/micro_structures.dmi'
+	icon_state = "mouse_hole"
+
+	anchored = TRUE
+	density = FALSE
+
+	var/max_accepted_scale = 0.3
+	var/magic = FALSE	//For events and stuff, if true, this tunnel will show up in the list regardless of whether it's in valid range, of if you're in a tunnel with this var, all tunnels of the same faction will show up redardless of range
+
+/obj/structure/micro_tunnel/Initialize()
+	. = ..()
+	if(name == "mouse hole")
```

Comment:
I think better way to do it is to make a variable called something like "autoname" and if its true, set our name upon init, otherwise skip, rather than checking if that specifically is our name.

Reply:
I don't think is necessary, either you name it yourself or it names itself after the area. When you go to select a destination, if there's things that are the same name, then it only puts that name on the list once, and picks one to go to at random. I'd prefer they get renamed than not.

Reply:
I am mostly proposing it for possibility of subtypes of this, or having different name but also adding area in front... Actually, in that regard, potentiall setting name to "[our_area.name] [name]" might work better. Still, its minor enough, just thought to suggest.